<a href="https://colab.research.google.com/github/IliyaPaulina/3mtt-IliyaP/blob/main/Week_6_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Apply advanced statistical techniques to a dataset presenting findings and insight.

HYPOTHESIS

To find wether or not the mean weight of a certain species of Dogs is equal to 310 pound.

In [ ]:
# importing scipy
import scipy.stats as stats

In [ ]:
# define dataset
data = [300, 315, 320, 311, 314, 309, 300, 308, 305, 303, 305, 301, 303]

In [ ]:
# perform one sample test
stats.ttest_1samp(a=data, popmean=310)

TtestResult(statistic=-1.5848116313861254, pvalue=0.1389944275158753, df=12)

The t test statistic is -1.5848 and the corresponding two sided p-value is0.1389.
This shows that
1. Null hypothesis is equal to 310 because the weigh of this specie is 310kg.
1. Alternative hypothesis is not equal to 310 because the mean weight is not equal to 310.




Chi-Square

To know if there is a relationship between the two categorical variables below.

In [ ]:
# impoting numpy
import numpy as np

In [ ]:
# importing scipy
from scipy.stats import chisquare

In [38]:
obs_data = [1, 2, 3, 4, 5, 6]
exp_data = [6, 5, 4, 3, 2, 1]

In [61]:
chisquare =(obs_data,exp_data)

Correlation Coefficient


In [ ]:
# importing  numpy
import numpy as np

In [54]:
# input array 1
a = np.arange(20, 30, 40)

In [59]:
#input array 2
b = np.array ([6, 7, 8])

In [ ]:
# calculating coefficient
x = np.corrcoef (a,b)
print(x)

there is a perfect negativ correlation  (-1.0) between a and b which means that when one vvariable increases, the other variables decreases in perfectly linear fashion.